src/random.h

```diff
@@ -67,8 +67,17 @@
  * Thread-safe.
  */
 void GetRandBytes(unsigned char* buf, int num) noexcept;
+/** Generate a random integer in the range [0..range). Precondition: range > 0 */
 uint64_t GetRand(uint64_t nMax) noexcept;
-std::chrono::microseconds GetRandMicros(std::chrono::microseconds duration_max) noexcept;
+/** Generate a random duration in the range [0..max). Precondition: max.count() > 0 */
+template <typename D>
+D GetRandomDuration(typename std::common_type<D>::type max) noexcept
```

Comment:
Perhaps add a comment here to explain why the `std::common_type` construction is used here.

Reply:
The idea came from here, btw: https://stackoverflow.com/a/37737627/8342274

Reply:
Done